File: init/erlang/messaging.erl
% messaging.erl
% 
-module(messaging).
-export([start/0, loop/0]).

start() -> spawn(fun() -> loop() end).

loop() ->
  receive
    {From, Msg} ->
      io:format("[~w] Got ~p from ~w~n",[self(), Msg, From]) ,
			From ! {self(), Msg} ,
			loop()
  end.

File: init/elixir/messaging.ex
# messaging.ex
#
defmodule Messaging do
  def start do spawn(fn -> loop end) end

  def loop do
    receive do
      {from, msg} ->
        pid = Kernel.inspect(self)
        from_pid = Kernel.inspect(from)
        IO.puts("[#{pid}] Got #{msg} from #{from_pid}")
  			send(from, {self, msg})
  			loop
    end
  end
end
